### PR TITLE
fix for crash (double free) when libnl3-devel (for HT channel support…

### PR DESCRIPTION
**Pull request by dmfreemon**
…) is not installed (fedora).

The configure script unsets CONFIG_LIBNL, which leads to the double free.

### CODE REVIEW

**Comment by dmfreemon** [comment]
Looking at the Travis output, I don't see what the problem might be.  make check works fine on my machine.  Please advise if there is something I should do for this.  Thanks.

**Comment by jbenden** [comment]
It may have been an intermittent thing, on Travis's part. The macOS image hung with no response after 10 minutes. I've restarted the job.

**Comment by dmfreemon** [comment]
Looking at the Travis output, I'm not sure I know what the problem is.  make check works fine on my machine.  But my machine is linux, and I think the build failure is on osx.

I do see this in Travis:
> checking whether build environment is sane... configure: error: newly created file is older than distributed files!
> Check your system clock

It's not clear if that is causing the non-zero return code.  But I am wondering if this could be indicating a problem with the build machine.

**Comment by mgaflip** [comment]
see https://github.com/aircrack-ng/aircrack-ng/issues/2062#issuecomment-597567304

File: lib/osdep/linux.c [hunk 1]
```diff
@@ -2259,7 +2257,7 @@ static int do_linux_open(struct wif * wi, char * iface)
 	dev->arptype_in = dev->arptype_out;
 
 	if (iface_malloced) free(iface);
-	if (iwpriv) free(iwpriv);
```
Comment (mgaflip) [inline lib/osdep/linux.c:2262]: this line should be deleted/comented as iwpriv points to a static char array

File: lib/osdep/linux.c [hunk 2]
```diff
@@ -2275,7 +2273,6 @@ static int do_linux_open(struct wif * wi, char * iface)
 	close(dev->fd_in);
 	if (acpi) fclose(acpi);
 	if (iface_malloced) free(iface);
-	if (iwpriv) free(iwpriv);
```
Comment (mgaflip) [inline lib/osdep/linux.c:2278]: this line should be deleted/commented as iwpriv points to a static char array

**Comment by dmfreemon** [comment]
@mgaflip Thanks for taking the time to share your comments above.  Can you confirm that you were _not_ using this PR when experiencing this crash and that your code change suggestions above are referring to the current master code and not code in this PR?

I believe the existing code in this PR has already fixed this bug.

**Comment by mcampana** [comment]
Hi, I'm using the master code. The big is there

**Comment by mcampana** [comment]
Bug, I mean

File: lib/osdep/linux.c (hunk 1 above)
Comment (RafaelLeeImg) [reply lib/osdep/linux.c:2262]: This line should be deleted/comented, agreed.
The iwpriv will not exist anymore after this patch merges. All stuff return by wiToolsPath should be free, wiToolsPath  uses malloc inside.

File: lib/osdep/linux.c (hunk 2 above)
Comment (RafaelLeeImg) [reply lib/osdep/linux.c:2278]: Same as above.
This line should be deleted/comented, agreed.
The iwpriv will not exist anymore after this patch merges. All stuff return by wiToolsPath should be free, wiToolsPath uses malloc inside.

**Comment by RafaelLeeImg** [comment]
Pretty good, this PR fixes the double double free error, works well on my Debian.
I confimed the base 491526f1e069328feacc54db7c7d82b62a2cb89c has double free problem.
I checked the modified lines, it solved the problem.
All dynamically allocated resources are freed in lib/osdep/linux.c, line 2279
```C
static void do_free(struct wif * wi)
```

**Comment by jbenden** [comment]
I have added missing pieces from this PR to current master.

Thanks!
-Joe

**Pull request closed without merging**